### PR TITLE
Jetpack Connect: Fetch only the connected site after authorize

### PR DESCRIPTION
**Pull request by seear**
Instead of all the sites! We need to force the API to fetch from wpcom instead of the .org site, which may not yet be accessible.

## Testing
* Check that a site can still be connected at /jetpack/connect, or from wp-admin
* The newly connected site should be present in the sites list in the calypso sidebar after connection
* Check the other connection flows listed at PCYsg-dek-p2

### CODE REVIEW

File: client/state/jetpack-connect/actions.js [hunk 1]
```diff
@@ -313,29 +313,22 @@ export function authorize( queryObject ) {
 				} );
 				// Update the user now that we are fully connected.
 				userFactory().fetch();
-				return wpcom.me().sites( {
-					site_visibility: 'all',
-					include_domain_only: true,
-					fields:
-						'ID,URL,name,capabilities,jetpack,visible,is_private,is_vip,icon,plan,jetpack_modules,single_user_site,is_multisite,options', //eslint-disable-line max-len
```
Comment (sirreal) [inline client/state/jetpack-connect/actions.js]: With this change, these fields are missing from this PR with respect to current behavior:

- `is_multisite`
- `is_private`
- `is_vip`
- `single_user_site`

And `meta` is included, where it wasn't before.

File: client/state/jetpack-connect/actions.js [hunk 2]
```diff
@@ -313,29 +313,22 @@ export function authorize( queryObject ) {
...
-					options:
-						'is_mapped_domain,unmapped_url,admin_url,is_redirect,is_automated_transfer,allowed_file_types,show_on_front,main_network_site,jetpack_version,software_version,default_post_format,created_at,frame_nonce,publicize_permanently_disabled,page_on_front,page_for_posts,advanced_seo_front_page_description,advanced_seo_title_formats,verification_services_codes,podcasting_archive,is_domain_only,default_sharing_status,default_likes_enabled,wordads,upgraded_filetypes_enabled,videopress_enabled,permalink_structure,gmt_offset,design_type', //eslint-disable-line max-len
```
Comment (sirreal) [inline client/state/jetpack-connect/actions.js]: These fields appear to be missing compared with current behavior:

- `advanced_seo_front_page_description`
- `default_sharing_status`
- `design_type`
- `gmt_offset`
- `is_automated_transfer`
- `is_domain_only`
- `is_multi_network`
- `is_multi_site`
- `is_redirect`
- `main_network_site`
- `max_upload_size`
- `podcasting_archive`
- `publicize_permanently_disabled`
- `verification_services_codes`
- `videopress_enabled`
- `wordads`

Comment (sirreal) [reply client/state/jetpack-connect/actions.js]: And these are included (weren't before):

- `ak_vp_bundle_enabled`
- `background_color`
- `blog_public`
- `default_category`
- `default_comment_status`
- `default_ping_status`
- `featured_images_enabled`
- `header_image`
- `headstart`
- `headstart_is_fresh`
- `image_default_link_type`
- `image_large_height`
- `image_large_width`
- `image_medium_height`
- `image_medium_width`
- `image_thumbnail_crop`
- `image_thumbnail_height`
- `image_thumbnail_width`
- `is_wpcom_store`
- `login_url`
- `post_formats`
- `site_goals`
- `theme_slug`
- `timezone`
- `woocommerce_is_active`

**Comment by seear** [comment]
> the query used here takes ~300ms. The query we're currently using in production takes ~13000ms. That's an huge difference

Excellent!

> I don't see any reason to remove the existing fields or options. They appear to be supported by the endpoint. Was there motivation in removing them?

1. Naive assumption that these options/fields had been added to cut down on the amount of data on the wire
1. It seemed to me to be both messy and fragile to specify all the fields and options we need

This would probably a better and less risky PR by just keeping the fields/options as the were. If we want to trim any or change this aspect we can do it in a different PR.

**Comment by seear** [comment]
> This would probably a better and less risky PR by just keeping the fields/options as they were.

(I'll add them back.)

**Comment by seear** [comment]
Ok, now keeping the original fields and options. In my testing this query takes 460ms compared to 12.9 seconds in prod.

**Comment by tyxla** [comment]
Thank you for this, @seear, this is still such a huge improvement!

File: client/state/jetpack-connect/actions.js [hunk 3]
```diff
@@ -328,14 +328,13 @@ export function authorize( queryObject ) {
...
-				debug( 'Sites list updated!', data );
+				debug( 'Site updated', data );
 				dispatch( {
-					type: SITES_RECEIVE,
-					sites: data.sites,
+					type: SITE_RECEIVE,
+					site: data,
 				} );
 				dispatch( {
 					type: JETPACK_CONNECT_AUTHORIZE_RECEIVE_SITE_LIST,
```
Comment (sirreal) [inline client/state/jetpack-connect/actions.js]: A nice follow-up will be to drop this action entirely, and just use `SITE_RECEIVE` in the reducer:

https://github.com/Automattic/wp-calypso/blob/d7fde50d0e211e98ff59d6fb1b1b1abadda1da90/client/state/jetpack-connect/reducer/jetpack-connect-authorize.js#L56-L60

**Comment by seear** [comment]
> An option we discussed elsewhere would be to roll this out "progressively" as an AB test where we can monitor the performance before rolling out for everyone. Do you think there's reason to be so cautious?

I love the idea of progressive roll out, and I've made changes before where I had wished we could do something like that.

However, I'm not sure that this change warrants that extra complexity, because:
1. As it stands I don't think this PR is too risky. We are replacing a fetch of all sites with a known-good way to fetch a single site.
1. Would a progressive roll out actually make it harder to spot any dips in the numbers?

So I suggest we merge when ready and keep an eye on the funnels.

**Comment by sirreal** [comment]
> So I suggest we merge when ready and keep an eye on the funnels.

I say ship!

**Pull request merged**